Clean up javadoc mojos execution

**BUG REPORT**

1. Please describe the issue you observed:

- What did you do?

run "mvn clean install -DskipTests" on current master

- What did you expect to see?
build okay without warnings

- What did you see instead?

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.0.0:javadoc (default-cli) on project bookkeeper-tools-framework: An error has occurred in Javadoc report generation: 
[ERROR] Exit code: 1 - javadoc: error - No public or protected classes found to document.

The error is hidden in the logs

during the build of bookkeeper-server module you can find this kind of lines:

> [INFO] --- maven-javadoc-plugin:3.0.0:jar (attach-javadocs) @ bookkeeper-server ---
> [INFO] The goal 'org.apache.maven.plugins:maven-javadoc-plugin:3.0.0:javadoc' has not been previously called for the module: 'org.apache.bookkeeper:bookkeeper-common:jar:4.8.0-SNAPSHOT'. Trying to invoke it...
> [WARNING] Creating fake javadoc directory to prevent repeated invocations: /xxxxx/bookkeeper/bookkeeper-common/target/apidocs
> [INFO] The goal 'org.apache.maven.plugins:maven-javadoc-plugin:3.0.0:javadoc' has not been previously called for the module: 'org.apache.bookkeeper:bookkeeper-proto:jar:4.8.0-SNAPSHOT'. Trying to invoke it...
> [ERROR] MavenInvocationException: Error when invoking Maven, consult the invoker log file: /home/diennea.lan/enrico.olivelli/dev/bookkeeper/bookkeeper-server/target/invoker/maven-javadoc-plugin1922966295.txt
> [WARNING] Creating fake javadoc directory to prevent repeated invocations: xxxxxx/bookkeeper/bookkeeper-proto/target/apidocs
> [INFO] The goal 'org.apache.maven.plugins:maven-javadoc-plugin:3.0.0:javadoc' has not been previously called for the module: 'org.apache.bookkeeper:bookkeeper-tools-framework:jar:4.8.0-SNAPSHOT'. Trying to invoke it...
> [ERROR] MavenInvocationException: Error when invoking Maven, consult the invoker log file: /home/diennea.lan/enrico.olivelli/dev/bookkeeper/bookkeeper-server/target/invoker/maven-javadoc-plugin952120288.txt
> [WARNING] Creating fake javadoc directory to prevent repeated invocations: /xxxxx/bookkeeper/tools/framework/target/apidocs
> [INFO] The goal 'org.apache.maven.plugins:maven-javadoc-plugin:3.0.0:javadoc' has not been previously called for the module: 'org.apache.bookkeeper.http:http-server:jar:4.8.0-SNAPSHOT'. Trying to invoke it...

then:

> [ERROR] MavenInvocationException: Error when invoking Maven, consult the invoker log file: /xxxxx/bookkeeper/bookkeeper-server/target/invoker/maven-javadoc-plugin1088547901.txt


now look into maven-javadoc-plugin1088547901.txt

@sijie

--- a/pom.xml
+++ b/pom.xml
@@ -907,7 +907,7 @@
         <artifactId>maven-javadoc-plugin</artifactId>
         <version>${maven-javadoc-plugin.version}</version>
         <configuration>
-          <additionalparam>-notimestamp</additionalparam>
+          <notimestamp>true</notimestamp>
           <!-- Prevent missing javadoc comments from being marked as errors -->
           <doclint>none</doclint>
           <subpackages>org.apache.bookkeeper.client:org.apache.bookkeeper.client.api:org.apache.bookkeeper.common.annotation:org.apache.bookkeeper.conf:org.apache.bookkeeper.feature:org.apache.bookkeeper.stats</subpackages>
@@ -936,6 +936,10 @@
           <overview>site/_site/overview/index.html</overview>
           <show>package</show>
           <detectJavaApiLink>false</detectJavaApiLink>
+          <!-- The javadoc plugin only runs in the javadoc modules. But with the default configuration it tries
+               to run a new maven instance in every dependency, in order to generate the apidocs there as well.
+               {@link https://maven.apache.org/plugins-archives/maven-javadoc-plugin-3.1.1/javadoc-mojo.html#detectOfflineLinks} -->
+          <detectOfflineLinks>false</detectOfflineLinks>
         </configuration>
         <executions>
           <execution>

